code/modules/mob/living/abilities/step_strike.dm

```diff
@@ -77,8 +77,16 @@
 
 		target_loc = pick(candidates)
 
-		if(!check_trajectory(target_loc, user, user.pass_flags))
-			return	//An obstacle prevents you from step striking. There is no reason to check the closest possible place to teleport since step strike can be used only if target is 2 tiles away or closer
+		var/sfat = TRUE //search_for_avalaibe_turfs
+		while(sfat)
+			if(!check_trajectory(target_loc, user, user.pass_flags|PASS_FLAG_TABLE|PASS_FLAG_NOMOB))
+				if(!candidates.len)
+					return //An obstacle prevents you from step striking. There is no reason to check the closest possible place to teleport since step strike can be used only if target is 2 tiles away or closer
```

Comment:
you've got this mixed up, its the charge that requires a minimum distance, step strike can be used to dance around someone you're already in melee with

Reply:
it works when the person is 2 tiles away and if there is window between you and your target you'll phase through it without this check